src/parser/jobs/rdm/modules/Gauge.tsx

```diff
@@ -0,0 +1,307 @@
+			color: JOBS.BLACK_MAGE.colour,
+		},
+		maximum: 100,
+		minimum: 0,
+	}))
+	public _gaugeModifiers = new Map<number, GaugeModifier>([
+		[this.data.actions.VERAERO.id, {white: 11, black: 0}],
+		[this.data.actions.VERAERO_II.id, {white: 7, black: 0}],
+		[this.data.actions.VERSTONE.id, {white: 9, black: 0}],
+		[this.data.actions.VERHOLY.id, {white: 21, black: 0}],
+		[this.data.actions.VERTHUNDER.id, {white: 0, black: 11}],
+		[this.data.actions.VERTHUNDER_II.id, {white: 0, black: 7}],
+		[this.data.actions.VERFIRE.id, {white: 0, black: 9}],
+		[this.data.actions.VERFLARE.id, {white: 0, black: 21}],
+		[this.data.actions.JOLT.id, {white: 3, black: 3}],
+		[this.data.actions.JOLT_II.id, {white: 3, black: 3}],
+		[this.data.actions.IMPACT.id, {white: 3, black: 3}],
+		[this.data.actions.SCORCH.id, {white: 7, black: 7}],
+	])
+	public _spenderModifiers = new Map<number, GaugeModifier>([
+		[this.data.actions.ENCHANTED_REPRISE.id, {white: -5, black: -5}],
+		[this.data.actions.ENCHANTED_MOULINET.id, {white: -20, black: -20}],
+		[this.data.actions.ENCHANTED_RIPOSTE.id, {white: -30, black: -30}],
+		[this.data.actions.ENCHANTED_ZWERCHHAU.id, {white: -25, black: -25}],
+		[this.data.actions.ENCHANTED_REDOUBLEMENT.id, {white: -25, black: -25}],
+	])
+	private _severityWastedMana = {
+		1: SEVERITY.MINOR,
+		20: SEVERITY.MEDIUM,
+		80: SEVERITY.MAJOR,
+	}
+	private _severityLostMana = {
+		1: SEVERITY.MINOR,
+		20: SEVERITY.MEDIUM,
+		80: SEVERITY.MAJOR,
+	}
+	public readonly _manaDifferenceThreshold = 30
+	private readonly _manaLostDivisor = 2
+	private readonly _manaficationMultiplier = 2
+	public readonly _manaCap = 100
+
+	_manaStatistics = {
+		white: {
+			manaficationLoss: 0,
+			imbalanceLoss: 0,
+			invulnLoss: 0,
+		},
+		black: {
+			manaficationLoss: 0,
+			imbalanceLoss: 0,
+			invulnLoss: 0,
+		},
+	}
+
+	override initialise() {
+		super.initialise()
+
+		this.addEventHook(
+			filter<Event>()
+				.type('damage')
+				.source(this.parser.actor.id),
+			this.onGaugeModifying
+		)
+		this.addEventHook(
+			filter<Event>()
+				.type('action')
+				.source(this.parser.actor.id)
+				.action(oneOf(Array.from(this._spenderModifiers.keys()))),
+			this.onGaugeSpender
+		)
+		this.addEventHook(
+			filter<Event>()
+				.type('action')
+				.source(this.parser.actor.id)
+				.action(this.data.actions.MANAFICATION.id),
+			this.onManafication
+		)
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	private onGaugeModifying(event: Events['damage']) {
+		if (event.cause.type !== 'action') {
+			return
+		}
+		const modifier = this._gaugeModifiers.get(event.cause.action)
+
+		if (modifier == null) {
+			return
+		}
+
+		const amount = modifier
+		const penalized = this.isOutOfBalance()
+		const whiteModified = penalized.white ? Math.ceil(amount.white/this._manaLostDivisor) : amount.white
+		const blackModified = penalized.black ? Math.ceil(amount.black/this._manaLostDivisor) : amount.black
```

Comment:
The game floors the mana gained when out of balance, not ceiling (using an 11 mana skill gains 5 mana of the low mana)
```suggestion
		const whiteModified = penalized.white ? Math.floor(amount.white/this._manaLostDivisor) : amount.white
		const blackModified = penalized.black ? Math.floor(amount.black/this._manaLostDivisor) : amount.black
```

Reply:
updated with suggestion.  Also I didn't realize I could have just clicked commit here heh.